src/Controller/Component/ModulesComponent.php

```diff
@@ -222,29 +222,78 @@ public function upload(array &$requestData) : void
             throw new \RuntimeException('Invalid form data: model-type');
         }
 
+        // has another stream? drop it
+        $this->removeStream($requestData);
+
         // upload file
         $filename = $requestData['file']['name'];
         $filepath = $requestData['file']['tmp_name'];
         $headers = ['Content-Type' => $requestData['file']['type']];
         $apiClient = ApiClientProvider::getApiClient();
         $response = $apiClient->upload($filename, $filepath, $headers);
 
-        // create media from stream
+        // assoc stream to media
         $streamId = $response['data']['id'];
+        $requestData['id'] = $this->assocStreamToMedia($streamId, $requestData, $filename);
```

Comment:
[memo] we probably need something in the API to replace stream without having to remove and add...